src/test/java/org/primefaces/component/terminal/TerminalAutoCompleteTest.java

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright 2009-2017 PrimeTek.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.primefaces.component.terminal;
+
+import static org.junit.Assert.*;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.primefaces.model.DefaultTreeNode;
+import org.primefaces.model.terminal.AutoCompleteMatches;
+
+public class TerminalAutoCompleteTest {
+
+    private Terminal terminal;
+
+    private DefaultTreeNode rootNode;
+
+    @Before
+    public void setup() {
+        terminal = new Terminal();
+
+        rootNode = new DefaultTreeNode("Root");
+
+        DefaultTreeNode show = new DefaultTreeNode("show");
+
+        DefaultTreeNode servers = new DefaultTreeNode("servers");
+        DefaultTreeNode status = new DefaultTreeNode("status");
+        DefaultTreeNode info = new DefaultTreeNode("info");
+
+        DefaultTreeNode version = new DefaultTreeNode("version");
+        DefaultTreeNode date = new DefaultTreeNode("date");
+
+        info.getChildren().add(version);
+        info.getChildren().add(date);
+
+        show.getChildren().addAll(Arrays.asList(servers, status, info));
+        rootNode.getChildren().add(show);
+    }
+
+    @After
+    public void teardown() {
+        terminal = null;
+        rootNode = null;
+    }
+
+    @Test
+    public void GIVEN_command_is_sho_WHEN_traverseCommandModel_THEN_returns_show_as_match() {
```

Comment:
Why not follow java method naming convention ?

Reply:
`given-when-then` unit test is nice, but the method should still be  in camel case IMHO (sorry for niche picking).

Reply:
It is a valid point, I did not check the conventions of the project, just automatically did it the way that I am used to ;) And as I am "a guest in the house", I made a commit which fixes the method names with camel case.

It is strange to the eye at first, but I find this convention more effective and readable for telling what the test does:

```java
GIVEN_command_is_show_s_WHEN_traverseCommandModel_THEN_returns_servers_and_status
```

versus:

```java
givenShowThenReturnsServersAndStatus
```

Reply:
> It is strange to the eye at first, but I find this convention more effective and readable for telling what the test does:

I see, but better to stick with camel case since we now have `checkstyle.xml` in the Primefaces project which can enforce camel checking if needed.

https://github.com/primefaces/primefaces/blob/master/checkstyle.xml

---

http://checkstyle.sourceforge.net/config_naming.html

> The Check validate abbreviations(consecutive capital letters) length in identifier name, it also allows to enforce camel case naming. Please read more at Google Style Guide to get to know how to avoid long abbreviations in names.